src/gui/guiFormSpecMenu.cpp

```diff
@@ -4169,13 +4313,22 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 std::string GUIFormSpecMenu::getNameByID(s32 id)
 {
 	for (FieldSpec &spec : m_fields) {
-		if (spec.fid == id) {
+		if (spec.fid == id)
 			return spec.fname;
-		}
 	}
 	return "";
 }
 
+
+FormspecFieldType GUIFormSpecMenu::getTypeByID(s32 id)
```

Comment:
This function could be merged with `getLabelByID`: `getFieldSpecById`, returning the entire fieldspec (or a default one).

Reply:
`getLabelByID` is inherited from `GUIModalMenu`:
https://github.com/minetest/minetest/blob/9519d57017cfa28eb4f2fbf3c780f826017bbb00/src/gui/modalMenu.h#L62-L63
See also: https://github.com/minetest/minetest/search?q=getLabelByID&unscoped_q=getLabelByID

Reply:
:man_shrugging: well then, OK.